tests/keras/engine/test_training.py

```diff
@@ -530,37 +531,218 @@ def test_model_with_input_feed_tensor():
+            return tf.train.Feature(bytes_list=tf.train.BytesList(value=[value]))
+
+        """ Save data into TFRecord """
+        if not os.path.isfile(filename):
+            num_examples = images.shape[0]
+
+            rows = images.shape[1]
+            cols = images.shape[2]
+            depth = images.shape[3]
+
+            print('Writing', filename)
+            writer = tf.python_io.TFRecordWriter(filename)
+            for index in range(num_examples):
+                image_raw = images[index].tostring()
+                example = tf.train.Example(features=tf.train.Features(feature={
+                    'height': _int64_feature(rows),
+                    'width': _int64_feature(cols),
+                    'depth': _int64_feature(depth),
+                    'label': _int64_feature(int(labels[index])),
+                    'image_raw': _bytes_feature(image_raw)}))
+                writer.write(example.SerializeToString())
+            writer.close()
+        else:
+            print('tfrecord %s already exists' % filename)
+
+    def read_and_decode_recordinput(tf_glob, one_hot=True, classes=None, is_train=None, batch_shape=[10, 3, 3, 1]):
+        """ Return tensor to read from TFRecord """
+        import tensorflow as tf
+        print('Creating graph for loading TFRecords...')
+        with tf.variable_scope("TFRecords"):
+            record_input = data_flow_ops.RecordInput(tf_glob, batch_size=batch_shape[0])
+            records_op = record_input.get_yield_op()
+            records_op = tf.split(records_op, batch_shape[0], 0)
+            records_op = [tf.reshape(record, []) for record in records_op]
+
+            images = []
+            labels = []
+            for i, serialized_example in enumerate(records_op):
+                with tf.variable_scope("parse_images", reuse=True):
+                    features = tf.parse_single_example(
+                        serialized_example,
+                        features={
+                            'label': tf.FixedLenFeature([], tf.int64),
+                            'image_raw': tf.FixedLenFeature([], tf.string),
+                        })
+                    img = tf.decode_raw(features['image_raw'], tf.uint8)
+                    img.set_shape(batch_shape[1] * batch_shape[2])
+                    img = tf.reshape(img, [1] + batch_shape[1:])
+
+                    img = tf.cast(img, tf.float32) * (1. / 255) - 0.5
+
+                    label = tf.cast(features['label'], tf.int32)
+                    if one_hot and classes:
+                        label = tf.one_hot(label, classes)
+
+                    images.append(img)
+                    labels.append(label)
+
+            images = tf.parallel_stack(images, 0)
+            labels = tf.parallel_stack(labels, 0)
+            images = tf.cast(images, tf.float32)
+
+            images = tf.reshape(images, shape=batch_shape)
+
+            return images, labels
+
+    def replace(filename):
```

Comment:
To test that functionality, disk tests are in many other places too like get_file. Im not aware of a way to create a tfrecord except on disk, but I haven't looked.